Microsoft.Research/Contracts/MsCorlib/System.Collections.ObjectModel.Collection.cs

```diff
@@ -0,0 +1,394 @@
+        //   item:
+        //     The object to insert. The value can be null for reference types.
+        //
+        // Exceptions:
+        //   System.ArgumentOutOfRangeException:
+        //     index is less than zero.-or-index is greater than System.Collections.ObjectModel.Collection<T>.Count.
+        protected virtual void InsertItem(int index, T item)
+        {
+            Contract.Requires((index >= 0) && (index <= Count));
+        }
+        //
+        // Summary:
+        //     Removes the first occurrence of a specific object from the System.Collections.ObjectModel.Collection<T>.
+        //
+        // Parameters:
+        //   item:
+        //     The object to remove from the System.Collections.ObjectModel.Collection<T>.
+        //     The value can be null for reference types.
+        //
+        // Returns:
+        //     true if item is successfully removed; otherwise, false. This method also
+        //     returns false if item was not found in the original System.Collections.ObjectModel.Collection<T>.
+        public bool Remove(T item)
+        {
+            return false;
+        }
+        //
+        // Summary:
+        //     Removes the element at the specified index of the System.Collections.ObjectModel.Collection<T>.
+        //
+        // Parameters:
+        //   index:
+        //     The zero-based index of the element to remove.
+        //
+        // Exceptions:
+        //   System.ArgumentOutOfRangeException:
+        //     index is less than zero.-or-index is equal to or greater than System.Collections.ObjectModel.Collection<T>.Count.
+        public void RemoveAt(int index) { }
+        //
+        // Summary:
+        //     Removes the element at the specified index of the System.Collections.ObjectModel.Collection<T>.
+        //
+        // Parameters:
+        //   index:
+        //     The zero-based index of the element to remove.
+        //
+        // Exceptions:
+        //   System.ArgumentOutOfRangeException:
+        //     index is less than zero.-or-index is equal to or greater than System.Collections.ObjectModel.Collection<T>.Count.
+        protected virtual void RemoveItem(int index)
+        {
+            Contract.Requires((index >= 0) && (index < Count));
+        }
+        //
+        // Summary:
+        //     Replaces the element at the specified index.
+        //
+        // Parameters:
+        //   index:
+        //     The zero-based index of the element to replace.
+        //
+        //   item:
+        //     The new value for the element at the specified index. The value can be null
+        //     for reference types.
+        //
+        // Exceptions:
+        //   System.ArgumentOutOfRangeException:
+        //     index is less than zero.-or-index is greater than System.Collections.ObjectModel.Collection<T>.Count.
+        protected virtual void SetItem(int index, T item)
+        {
+            Contract.Requires((index >= 0) && (index <= Count));
```

Comment:
:exclamation: This is incorrect. It should be `index < Count` instead of `index <= Count` (the documentation above it needs to be corrected as well.